src/components/config/parse-text.ts

```diff
@@ -0,0 +1,100 @@
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { Config } from '../../interfaces/config'
+import { DEFAULT_CONFIG_INTERVAL } from '.'
+import { DEFAULT_THRESHOLD } from '../../looper'
+import { Probe } from '../../interfaces/probe'
+import { isValidURL } from '../../utils/is-valid-url'
+
+export const parseConfigFromText = (configString: string): Config => {
+  try {
+    let probes: Probe[] = []
+    const urls = configString.split(/\r?\n/)
+    for (const [, url] of urls.entries()) {
+      if (isValidURL(url)) {
+        probes = [
+          ...probes,
+          {
+            id: url,
+            name: url,
+            requests: [
+              {
+                url: url,
+                method: 'GET',
+                timeout: 10_000,
+                body: {} as JSON,
+              },
+            ],
+            incidentThreshold: DEFAULT_THRESHOLD,
+            recoveryThreshold: DEFAULT_THRESHOLD,
+            interval: DEFAULT_CONFIG_INTERVAL,
+            alerts: [
+              {
+                assertion: 'response.status < 200 or response.status > 299',
+                message: 'HTTP Status is not 200',
+              },
+              {
+                assertion: 'response.time > 2000',
+                message: 'Response time is more than 2000ms',
+              },
+            ],
+          },
+        ]
+      }
+      // Ignoring new line from url validating checker
+      else if (url.length > 0) {
+        throw new Error('URL invalid')
+      }
+    }
+
+    return { probes }
+  } catch {
+    throw new Error('Your Text file contains an invalid format !')
+  }
```

Comment:
Do we need to catch it here if it's gonna throw another Error again?

Reply:
try catch removed @nicnocquee